### PR TITLE
Update to support 2.13 and Play 2.7.x

### PR DESCRIPTION
**Pull request by AprilAtProtenus**
Update project to support Scala 2.11 through 2.13.

Update project to support Play Framework 2.7.x.

Update CI configuration and add automatic CI publishing.

Add type annotations to public members and implicits.

### CODE REVIEW

File: src/main/scala/scaldi/play/ScaldiApplicationBuilder.scala [hunk 1]
```diff
@@ -126,8 +129,13 @@ object ScaldiApplicationBuilder {
                        modules: Seq[CanBeScaldiInjector] = Seq.empty,
                        disabled: Seq[Class[_]] = Seq.empty,
                        loadConfiguration: Environment => Configuration = Configuration.load,
-                       loadModules: (Environment, Configuration) => Seq[CanBeScaldiInjector] = ScaldiBuilder.loadModules)(fn: => T) = {
-    val app = new ScaldiApplicationBuilder(environment, configuration, modules, disabled, loadConfiguration, loadModules).build()
+                       loadModules: (Environment, Configuration) => Seq[CanBeScaldiInjector] = ScaldiBuilder.loadModules)(fn: => T): T = {
+    val app = new ScaldiApplicationBuilder(environment, configuration, modules, disabled, loadConfiguration, loadModules)
+      .appendModule(new Module {
+        // `play.api.ApplicationLoader.DevContext` not available without a `play.api.ApplicationLoader.Context`
+        bind [OptionalDevContext] to new OptionalDevContext(None)
```
Comment (AprilAtProtenus) [inline src/main/scala/scaldi/play/ScaldiApplicationBuilder.scala]: I don't feel confident enough in this change and why it should be different than for Play 2.6.x to just do a release, so I think I may do a milestone-type release (`0.6.0-M1`) and then try to test it with a small app to make sure it works.

File: build.sbt [hunk 2]
```diff
@@ -59,9 +49,9 @@ shellPrompt in ThisBuild := { state =>
...
-organizationHomepage := Some(url("https://github.com/scaldi"))
-developers := Developer("OlegIlyenko", "Oleg Ilyenko", "", url("https://github.com/OlegIlyenko")) :: Nil
+organizationHomepage := Some(url("https://github.com/protenus"))
+developers := Developer("AprilAtProtenus", "April Hyacinth", "", url("https://github.com/AprilAtProtenus")) :: Nil
 scmInfo := Some(ScmInfo(
-  browseUrl = url("https://github.com/scaldi/scaldi-play"),
-  connection = "scm:git:git@github.com:scaldi/scaldi-play.git"
+  browseUrl = url("https://github.com/protenus/dipendi-play"),
+  connection = "scm:git:git@github.com:protenus/dipendi-play.git"
 ))
```
Comment (jeffSweeney812) [inline build.sbt]: I know it was already like this, but why does it seem to care/warn about no newline being present here? Is it necessary to include it?

Comment (AprilAtProtenus) [reply build.sbt]: basically, some *nix tools and processes (see also the C compiler) don't handle the lack of a newline at the end of the file well. modern versions of most tools (e.g. `less`, `vim`) don't care. so it's better to have newlines at the end of the file, but it also doesn't really matter.

File: build.sbt [hunk 3]
```diff
@@ -1,54 +1,44 @@
-name := "scaldi-play"
-organization := "org.scaldi"
-version := "0.5.18-SNAPSHOT"
```
Comment (doughnutbox) [inline build.sbt]: dropping this is covered by the updated publishing?

Comment (AprilAtProtenus) [reply build.sbt]: correct

File: build.sbt (hunk 2 above)
Comment (jeffSweeney812) [reply build.sbt]: Ah I see. Fair enough.

File: build.sbt (hunk 3 above)
Comment (AprilAtProtenus) [reply build.sbt]: see https://github.com/olafurpg/sbt-ci-release#sbt

File: build.sbt (hunk 2 above)
Comment (AprilAtProtenus) [reply build.sbt]: see https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file for better and slightly more correct (not from my memory) detail

**Comment by AprilAtProtenus** [comment]
forgot to squash before merge, oops

**Comment by AprilAtProtenus** [comment]
fixed merge in d8b039e2012976b788f7fffb95c315e1c9961836

**Pull request merged**